File: absolute_difference.cpp
#include <bits/stdc++.h>
#include <cassert>
using namespace std;

//time  --> o(nlogn)
//space --> o(n)
int main()
{
    int n, ans = 1e9;
    cin >> n;
    assert((n >= 2) && "The size of the array must be greater than or equal to 2");
    vector<int> v(n);
    for (int i = 0; i < n; i++)
        cin >> v[i];

    sort(v.begin(), v.end());

    for (int i = 1; i < n; i++)
    {
        ans = min(ans, (v[i] - v[i - 1]));
    }
    cout << ans;
    return 0;
}

File: absolute_difference.py
def find_min_absolute_difference(ls,n):
    """
    time  --> o(nlogn)
    space --> o(n)
    """
    ls.sort()
    ans=1*10**9
    for i in range(1,n):
        ans =min(ans,(ls[i]-ls[i-1]))
    return ans



if __name__ == '__main__':
    n=int(input())
    assert n >=2,"The size of the array must be greater than or equal to 2";
    ls=list(map(int,input().split()))
    print(find_min_absolute_difference(ls,n))
    #print(find_min_absolute_difference.__doc__)



